flow/project.py

```diff
@@ -281,14 +266,12 @@ def __init__(self, name, job, cmd, directives=None, np=None):
 
         # Handle deprecated np argument:
         if np is not None:
-            warnings.warn(
-                "The np argument for the JobOperation constructor is deprecated.",
-                DeprecationWarning)
-            assert directives.setdefault('np', np) == np
-        else:
-            # Future: We can remove everything but the else: clause.
-            directives.setdefault(
-                'np', directives.get('nranks', 1) * directives.get('omp_num_threads', 1))
+            raise RuntimeError(
+                "The np argument for the JobOperation constructor has been deprecated "
+                "as of version 0.6 and been removed as of version 0.7!")
```

Comment:
I'm confused why this is an error, whereas most of the "will be removed in 0.7" (e.g. the `deprecated_since_06` decorator) is just a warning.